submissions/YuliiaDyka/dom-api/base.js

```diff
@@ -0,0 +1,221 @@
+"use strict"
+
+
+const dataObj = {
+    'Git Basics':`<ol class="task">
+    <li class="task__item">
+        Прослухайте тижні 1 і 2 курсу <a class="link tomato" href="https://www.coursera.org/learn/introduction-git-github">Introduction to Git and GitHub</a> 
+    </li>
+    <li class="task__item">
+        Пройдіть наступні рівні тут <a class="link tomato" href="https://learngitbranching.js.org/">learngitbranching.js.org:</a>
+        <ul>
+            <li class="task__item">Основи: Introduction Sequence, Ramping Up</li>
+            <li class="task__item">Віддалені репозиторії: Push & Pull -- віддалені репозиторії в Git!</li>
+        </ul>
+    </li>
+    <li class="task__item">
+        Створіть репозиторій та назвіть його kottans-frontend.
+    </li>
+    <li class="task__item">
+        Створіть README.md для репозиторію.
+    </li>
+    <li class="task__item">
+        Опишіть свої враження від вивченого матеріалу.
+    </li>
+    <li class="task__item">
+        Надішліть pull-request сюди Kottans/mock-repo пропонуючи зміни.
+    </li>
+ </ol>`,
```

Comment:
Avoid a tag name in the data list.
Get more interactive objects. 
```javascript
const taskList = [{
       id: 0,
       taskName: 'Git Basics',
       taskExesice: ['Опишіть свої враження від вивченого матеріалу.', 'Надішліть pull-request сюди Kottans/mock-repo пропонуючи зміни.', ....... ],
       linkHomePage: '.....',
       linkTaskPage: '.......'
}, ....]
```

Reply:
Hello) I fixed dataObj, but left the link in list items, because that was the only way I could keep the links to certain words in the description of the task

Reply:
```javascript
const dataObj = [
    {
        taskName: 'Git Basics',
        taskExesice: [   
            {
                url: "https://www.coursera.org/learn/introduction-git-github",
                urlText: "Introduction to Git and GitHub",
                taskText: null
            },   
            {
                url: "https://learngitbranching.js.org/",
                urlText: "learngitbranching.js.org",
                taskText: "Пройдіть наступні рівні тут "
            },   
            {
                url: null,
                urlText: null,
                taskText: "Створіть репозиторій та назвіть його kottans-frontend."
            },
            {
                url: null,
                urlText: null,
                taskText: "Створіть README.md для репозиторію."
            },
            {
                url: null,
                urlText: null,
                taskText: "Опишіть свої враження від вивченого матеріалу."
            },
            {
                url: null,
                urlText: null,
                taskText: "Надішліть pull-request сюди Kottans/mock-repo пропонуючи зміни."
            },
        ]
    }, 
    
    {
        taskName: 'Linux CLI and Networking',
        taskExesice: [   
            {
                url: "https://linuxsurvival.com/linux-tutorial-introduction/",
                urlText: "Linux Survival (4 modules)",
                taskText: null
            },   
            {
                url: "https://code.tutsplus.com/uk/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177",
                urlText: "HTTP: Протокол, який повинен розуміти кожний веб-розробник - Частина 1",
                taskText: null
            },   
            {
                url: "https://code.tutsplus.com/uk/tutorials/http-the-protocol-every-web-developer-must-know-part-2--net-31155",
                urlText: "HTTP: Протокол, який повинен розуміти кожний веб-розробник - Частина 2",
                taskText: null
            },
        ]
    },
......
];

function renderPage(targetTaskName) {
    const desc = document.createElement('div');
    desc.classList.add('description');
    const taskList = document.createElement("ol");
    taskList.classList.add('task');

    const filteredTaskList = dataObj.filter(task => task.taskName === targetTaskName);
    const [{ taskExesice }] = filteredTaskList;

    const taskExesiceList = taskExesice.map(task => {
        const { taskText, url, urlText } = task;
        return`
            <li class="task__item">
                ${taskText ? taskText : ""}
                ${url ? `<a class="link tomato" href="${url}" target="_blank">${urlText}</a>` : ""}
            </li>
        `;
    }).join('');

    taskList.innerHTML = taskExesiceList;
    desc.append(taskList);
    return desc;
}
```

Reply:
@madmaxWMFU thank you! Its works!)))